super-ddf.c

```diff
@@ -3497,19 +3510,22 @@ static int load_super_ddf_all(struct supertype *st, int fd,
 	if (sra->array.major_version != -1 ||
 	    sra->array.minor_version != -2 ||
 	    strcmp(sra->text_version, "ddf") != 0)
-		return 1;
+		goto error1;
```

Comment:
please do not use error1, there are better ways do set error code, see my proposal.